net-libs/mbedtls/mbedtls-2.6.0.recipe

```diff
@@ -0,0 +1,126 @@
+	mbedtls$secondaryArchSuffix = $portVersion
+	lib:libmbedcrypto$secondaryArchSuffix = 0 compat >=0
+	lib:libmbedtls$secondaryArchSuffix = 10 compat >= 10
+	lib:libmbedx509$secondaryArchSuffix = 0 compat >= 0
+	"
+if [ -z "$secondaryArchSuffix" ]; then
+	PROVIDES="$PROVIDES
+		cmd:mbedtls_aescrypt2 = $portVersion
+		cmd:mbedtls_benchmark = $portVersion
+		cmd:mbedtls_cert_app = $portVersion
+		cmd:mbedtls_cert_req = $portVersion
+		cmd:mbedtls_cert_write = $portVersion
+		cmd:mbedtls_crl_app = $portVersion
+		cmd:mbedtls_crypt_and_hash = $portVersion
+		cmd:mbedtls_dh_client = $portVersion
+		cmd:mbedtls_dh_genprime = $portVersion
+		cmd:mbedtls_dh_server = $portVersion
+		cmd:mbedtls_dtls_client = $portVersion
+		cmd:mbedtls_dtls_server = $portVersion
+		cmd:mbedtls_ecdh_curve25519 = $portVersion
+		cmd:mbedtls_ecdsa = $portVersion
+		cmd:mbedtls_gen_entropy = $portVersion
+		cmd:mbedtls_gen_key = $portVersion
+		cmd:mbedtls_gen_random_ctr_drbg = $portVersion
+		cmd:mbedtls_gen_random_havege = $portVersion
+		cmd:mbedtls_generic_sum = $portVersion
+		cmd:mbedtls_hello = $portVersion
+		cmd:mbedtls_key_app = $portVersion
+		cmd:mbedtls_key_app_writer = $portVersion
+		cmd:mbedtls_mini_client = $portVersion
+		cmd:mbedtls_mpi_demo = $portVersion
+		cmd:mbedtls_pem2der = $portVersion
+		cmd:mbedtls_pk_decrypt = $portVersion
+		cmd:mbedtls_pk_encrypt = $portVersion
+		cmd:mbedtls_pk_sign = $portVersion
+		cmd:mbedtls_pk_verify = $portVersion
+		cmd:mbedtls_req_app = $portVersion
+		cmd:mbedtls_rsa_decrypt = $portVersion
+		cmd:mbedtls_rsa_encrypt = $portVersion
+		cmd:mbedtls_rsa_genkey = $portVersion
+		cmd:mbedtls_rsa_sign = $portVersion
+		cmd:mbedtls_rsa_sign_pss = $portVersion
+		cmd:mbedtls_rsa_verify = $portVersion
+		cmd:mbedtls_rsa_verify_pss = $portVersion
+		cmd:mbedtls_selftest = $portVersion
+		cmd:mbedtls_ssl_cert_test = $portVersion
+		cmd:mbedtls_ssl_client1 = $portVersion
+		cmd:mbedtls_ssl_client2 = $portVersion
+		cmd:mbedtls_ssl_fork_server = $portVersion
+		cmd:mbedtls_ssl_mail_client = $portVersion
+		cmd:mbedtls_ssl_server = $portVersion
+		cmd:mbedtls_ssl_server2 = $portVersion
+		cmd:mbedtls_strerror = $portVersion
+		cmd:mbedtls_udp_proxy = $portVersion
+		"
+fi
+REQUIRES="
+	haiku$secondaryArchSuffix
+	"
+
+PROVIDES_devel="
+	mbedtls${secondaryArchSuffix}_devel = $portVersion
+	devel:libmbedcrypto$secondaryArchSuffix = 0 compat >=0
+	devel:libmbedtls$secondaryArchSuffix = 10 compat >= 10
+	devel:libmbedx509$secondaryArchSuffix = 0 compat >=0
+	"
+REQUIRES_devel="
+	mbedtls$secondaryArchSuffix == $portVersion base
+	"
+
+BUILD_REQUIRES="
+	haiku${secondaryArchSuffix}_devel
+	"
+BUILD_PREREQUIRES="
+	cmd:gcc$secondaryArchSuffix
+	cmd:make
+	cmd:perl
+	cmd:sed
+	"
+
+PATCH()
+{
+	sed -i \
+		-e "s|/include/mbedtls|/$relativeOldIncludeDir|;"\
+		-e "s|/include|/$relativeOldIncludeDir|;"\
+		-e "s|/bin/|/$relativeBinDir/|;"\
+		-e "s|/bin\$|/$relativeBinDir/|;"\
+		-e "s|/lib\$|/$relativeLibDir/|;"\
+		Makefile
+}
+
+BUILD()
```

Comment:
Here's what we can insert, before BUILD, to have the debuginfo packages:
```
if [ -n "$secondaryArchSuffix" ]; then
	maybe_binDir_mbedtls_runtimes=
else
	maybe_binDir_mbedtls_runtimes="`echo "$PROVIDES" | sed -n \
		-e "s|cmd:\([^ ]*\).*|$binDir/\1|p" \
		`"
fi
defineDebugInfoPackage mbedtls$secondaryArchSuffix \
	$libDir/libmbedcrypto.so.0 \
	$libDir/libmbedtls.so.10 \
	$libDir/libmbedx509.so.0 \
	$maybe_binDir_mbedtls_runtimes

```
(Make sure you keep an empty line before BUILD.)

I'm giving the trick here because it is not easy to figure out how to handle the debuginfo package for all cases (i.e. primary vs secondary arch builds), especially with 47 runtimes in PROVIDES.
We usually list them explicitly, but it is much better to compute (from `$PROVIDES`) the arguments we need to pass to defineDebugInfoPackage.

Reply:
All my apologies. I wrote "before BUILD" but I should have written "before PATCH".
The fix is easy to do with ed:
```
(echo 96,106 m 119; echo w; echo q) | ed mbedtls-2.6.0.recipe
```
I assumes the PATH function starts at line 96 and that the BUILD function is at line 120.
I will move the 11 lines from 96 to 106 (i.e. the bloc containing the PATCH function and the blank line after it) a few lines below, just before BUILD.